src/power_grid_model/validation/validation.py

```diff
@@ -331,7 +379,25 @@ def validate_required_values(
         required["line"] += ["r0", "x0", "c0", "tan0"]
         required["shunt"] += ["g0", "b0"]
 
-    return list(chain(*(none_missing(data, component, required.get(component, [])) for component in data)))
+    # Mark `power_sigma` for each power_sensor based on the state of `p_sigma` and `q_sigma`
+    process_power_sigma_and_p_q_sigma(data, "sym_power_sensor", required)
+    process_power_sigma_and_p_q_sigma(data, "asym_power_sensor", required)
+
+    return list(
+        chain(
+            *(
+                none_missing(data, component, item, index)
+                for component in data
+                for index, item in enumerate(
+                    (sublist for sublist in required.get(component, []))
+                    if isinstance(required.get(component, []), list)
+                    and all(isinstance(i, list) for i in required.get(component, []))
+                    else [required[component]]
+                )
+                if data[component] is not None and isinstance(data[component], Sized) and index < len(data[component])
```

Comment:
@mgovers @Jerry-Jinfeng-Guo decision after discussion:

Clean this part up to make it more readable and less intrusive.

Reply:
Code cleaned in the latest commit.